packages/mask/src/plugins/Profile/SNSAdaptor/DonationsPage.tsx

```diff
@@ -1,10 +1,110 @@
+import { useState, useEffect } from 'react'
+import DonationCard from './components/DonationCard'
+import type { GeneralAssetWithTags } from './common/types'
 import { Typography } from '@mui/material'
+import config from './common/config'
+import Button from './components/Button'
+import { COLORS } from './common/variables'
+import { makeStyles } from '@masknet/theme'
+import formatter from './common/address'
+import { CircularProgress } from '@mui/material'
+import utils from './common/utils'
 
-interface DonationPageProps {}
+const useStyles = makeStyles()((theme) => ({
+    msg: {
+        color: theme.palette.mode === 'dark' ? 'rgb(255, 255, 255)' : 'rgb(0, 0, 0)',
+    },
+    primaryText: {
+        color: theme.palette.primary.main,
+    },
+}))
+interface DonationPageProps {
+    address: string
+    isOwnAddress: boolean
+    isConnected: boolean
+}
 export function DonationPage(props: DonationPageProps) {
+    const { address, isOwnAddress, isConnected } = props
+    const { classes } = useStyles()
+
+    const [listedDonation, setlistedDonation] = useState<GeneralAssetWithTags[]>([])
+    const [isLoading, setIsLoading] = useState(true)
+
+    const toSingleDonation = (platform: string, identity: string, id: string, type: string) => {
+        window.open(`https://rss3.bio/${address}/singlegitcoin/${platform}/${identity}/${id}/${type}`)
```

Comment:
```suggestion
        window.open(`https://rss3.bio/${address}/singlegitcoin/${platform}/${identity}/${id}/${type}`, '_blank', 'noopener noreferrer')
```

Reply:
@azuic plz use https://npm.im/urlcat build the url.